packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  Attachment,
+  Bundle,
+  CodeableConcept,
+  Coding,
+  DocumentReference,
+  Extension,
+  Identifier,
+} from "@medplum/fhirtypes";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { cloneDeep } from "lodash";
+import { MedicalDataSource, isMedicalDataSource } from "..";
+import { createAttachmentUploadPath } from "../../domain/document/upload";
+import { createFilePathFromFileName } from "../../domain/filename";
+import { executeAsynchronously } from "../../util/concurrency";
+import { Config } from "../../util/config";
+import { out } from "../../util/log";
+import { sizeInBytes } from "../../util/string";
+import { uuidv4 } from "../../util/uuid-v7";
+import { S3Utils, UploadParams } from "../aws/s3";
+import { cqExtension } from "../carequality/extension";
+import { cwExtension } from "../commonwell/extension";
+import { FhirClient } from "../fhir/api/api";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const s3Utils = new S3Utils(Config.getAWSRegion());
+const s3BucketName = Config.getMedicalDocumentsBucketName();
+
+export async function processAttachments({
+  b64Attachments,
+  cxId,
+  patientId,
+  fileName,
+  medicalDataSource,
+  fhirApi,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  b64Attachments: any[];
+  cxId: string;
+  patientId: string;
+  fileName: string;
+  medicalDataSource?: string | undefined;
+  fhirApi?: FhirClient;
+}) {
+  const { log } = out(`processAttachments - cxId ${cxId}, patientId: ${patientId}`);
+  const filePath = createFilePathFromFileName(fileName);
+
+  const extensions = [buildDocIdFhirExtension(filePath), getSourceExtension(medicalDataSource)]
+    .flat()
+    .filter(Boolean) as Extension[];
+
+  const initialDocRef: DocumentReference = {
+    resourceType: "DocumentReference",
+    status: "current",
+    docStatus: "final",
+    subject: {
+      reference: `Patient/${patientId}`,
+      type: "Patient",
+    },
+    extension: extensions,
+  };
+
+  const docRefs: DocumentReference[] = [];
+  const uploadDetails: UploadParams[] = [];
+
+  b64Attachments.map(async att => {
+    const act = att.act;
+
+    const fileDetails = getDetails(act.text);
+    const docRef = fillDocumentReference(initialDocRef, act);
+    if (!docRef.id) throw new Error("Missing ID in DocRef");
```

Comment:
this can't be undefined based on the code...

Reply:
ya, but we're using FHIR types from medplum, and resource IDs are always optional. I was thinking to use an assertion operator (`!`), but decided to use an error instead. Lmk what's a better approach here in your opinion